projects/plugins/jetpack/extensions/plugins/launchpad-save-modal/index.js

```diff
@@ -0,0 +1,82 @@
+/* global launchpadModalOptions */
+
+import { Modal, Button, CheckboxControl } from '@wordpress/components';
+import { usePrevious } from '@wordpress/compose';
+import { useSelect } from '@wordpress/data';
+import { store as editorStore } from '@wordpress/editor';
+import { __ } from '@wordpress/i18n';
+import { useEffect, useState } from 'react';
+import './editor.scss';
+
+export const name = 'launchpad-save-modal';
+
+export const settings = {
+	render: function LaunchpadSaveModal() {
+		const isSaving = useSelect(
+			select => select( editorStore ).isSavingNonPostEntityChanges(),
+			[]
+		);
+		const prevIsSaving = usePrevious( isSaving );
+		const [ isModalOpen, setIsModalOpen ] = useState( false );
+		const [ dontShowAgain, setDontShowAgain ] = useState( false );
+		const [ , siteSlug ] = launchpadModalOptions.siteUrlOption.split( '//' );
+		const isInsideSiteEditor = window.location.href.includes( 'site-editor' );
+
+		useEffect( () => {
+			if ( prevIsSaving === true && isSaving === false ) {
+				setIsModalOpen( true );
+			}
+		}, [ isSaving, prevIsSaving ] );
+
+		const showModal =
+			launchpadModalOptions.launchpadScreenOption === 'full' &&
+			launchpadModalOptions.siteIntentOption === 'link-in-bio' &&
+			isInsideSiteEditor &&
+			isModalOpen;
```

Comment:
```suggestion
		const showModal =
			launchpadModalOptions.launchpadScreenOption === 'full' &&
			launchpadModalOptions.siteIntentOption === 'link-in-bio' &&
			isInsideSiteEditor &&
			isModalOpen &&
			! dontShowAgain;
```
It looks like we have `useState` and checkbox toggle both set up for `dontShowAgain/setDontShowAgain`. But the modal does show again, I imagine this suggestion might be what we were missing?